src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -284,52 +300,88 @@ protected override void StopProcessing()
         #region ConnectionTest
 
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
-        {
+        {    
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
+            int successfulConnections = 0;
 
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
-
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                TcpTestStatus testResult = new TcpTestStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress.ToString(),
+                    TcpPort,
+                    0,
+                    TcpTestStatus.TcpConnectionTestResult.New
+                );
+                
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
+                    TcpClient client = new TcpClient();
+                    
+                    Stopwatch stopwatch = new Stopwatch();
+                    stopwatch.Start();
+
+                    Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
+                    Task timeoutTask = Task.Delay(new TimeSpan(0, 0, TimeoutSeconds));
+
                     Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    stopwatch.Stop();
+
+                    client.Close();
```

Comment:
We could remove this if we do above:
```
using TcpClient client = new TcpClient();
```

Reply:
Wrapped the TcpClient in a using statement, thanks!